src/IR/Simplify.hs

```diff
@@ -0,0 +1,459 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
```

Comment:
What's this for?

Reply:
We didn't need any of them except DerivingVia; thanks!